### PR TITLE
User script return & init eeprom correction

### PR DESCRIPTION
**Pull request by Tannoo**
Introducing an option to return to the status screen after each user script has completed.

Also added M501 to the Init eeprom LCD menu item. This allows UBL to be initialized correctly.

### CODE REVIEW

File: Marlin/src/lcd/ultralcd.cpp [hunk 1]
```diff
@@ -2997,7 +3000,7 @@ void kill_screen(const char* lcd_msg) {
     #endif
     MENU_ITEM(function, MSG_RESTORE_FAILSAFE, lcd_factory_settings);
     #if ENABLED(EEPROM_SETTINGS)
-      MENU_ITEM(gcode, MSG_INIT_EEPROM, PSTR("M502\nM500")); // TODO: Add "Are You Sure?" step
+      MENU_ITEM(gcode, MSG_INIT_EEPROM, PSTR("M502\nM500\nM501")); // TODO: Add "Are You Sure?" step
```
Comment (thinkyhead) [inline Marlin/src/lcd/ultralcd.cpp:3003]: If `M501` is required, then there must be a bug in `M502`. It should make no difference to simply read back what was just written.

Comment (Roxy-3D) [reply Marlin/src/lcd/ultralcd.cpp:3003]: M501 has always been required to get correct initialization.   That doesn't make sense, but historically, that is true.  I think it does indicate a bug in M502 or M500.

**Comment by thinkyhead** [comment]
Ready for merge at #7771 and #7772

File: Marlin/src/lcd/ultralcd.cpp (hunk 1 above)
Comment (thinkyhead) [reply Marlin/src/lcd/ultralcd.cpp:3003]: With so many changes to the codebase, including bug-fixes to the EEPROM code over the last several months, the historical kludge may no longer be needed. Both `M502` and `M501` write into the same memory slots, and both call the same (new-ish) method to recalculate derived values based on those values. I'll do a careful audit and see whether `M501` does anything special compared to `M502`.

**Comment by Tannoo** [comment]
With just M502 then M500, UBL cannot use the eeprom.
If I reset Marlin, then it's usable.

Or, M502, M500, then M501... then it's usable without a restart.

**Pull request closed without merging**